libc-bottom-half/crt/crt1-command.c

```diff
@@ -3,17 +3,21 @@ extern void __wasm_call_ctors(void);
 extern int __main_void(void);
 extern void __wasm_call_dtors(void);
 
-// Commands should only be called once per instance. This simple check ensures
-// that the `_start` function isn't started more than once.
-static volatile int started = 0;
-
 __attribute__((export_name("_start")))
 void _start(void) {
-    // Don't allow the program to be called multiple times.
+    // Commands should only be called once per instance. This simple check
+    // ensures that the `_start` function isn't started more than once.
+    static volatile int started = 0;
+#ifdef _REENTRANT
+    if (__sync_val_compare_and_swap(&started, 0, 1) != 0) {
```

Comment:
What about using C11 atomics here?  (https://www.ibm.com/docs/en/zos/2.1.0?topic=c11-atomic-compare-exchange-strong).  Maybe we don't want to depend on C11?